Fix for issue #245 - ATT_OP_FIND_INFO_RESP incorrect processing durin…

carterd opened the pull request:
Fix for issue #245 - ATT_OP_FIND_INFO_RESP incorrect processing during ATTClass::discoverDescriptors causes crashing

I've highlighted this issue on 9th July ... this is an issue of causing __CRASHES__ if using ArduinoBLE to connect as central and the response to ATT_OP_FIND_INFO_RESP includes 128-bit UUIDs.

Please check and remove comments after review

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino-libraries/ArduinoBLE?pullRequest=263) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino-libraries/ArduinoBLE?pullRequest=263) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino-libraries/ArduinoBLE?pullRequest=263) it.</sub>

carterd committed:
Fix for issue #245 - ATT_OP_FIND_INFO_RESP incorrect processing during ATTClass::discoverDescriptors causes crashing

I've highlighted this issue on 9th July ... this is an issue of causing __CRASHES__ if using ArduinoBLE to connect as central and the response to ATT_OP_FIND_INFO_RESP includes 128-bit UUIDs.

facchinm commented:
Hi @carterd ,
thanks for looking into this! I think the PR can be simplified a bit, I'm adding some inline comments. Will be more than happy to merge afterwards :wink:

facchinm commented on src/utility/ATT.cpp:
```diff
@@ -1726,8 +1726,19 @@ bool ATTClass::discoverDescriptors(uint16_t connectionHandle, BLERemoteDevice* d
+          //
+          // See section 3.4.3.2 ATT_FIND_INFORMATION_RSP of Bluetooth Core Specification 5.3.
+          //
+          uint16_t lengthPerDescriptor = responseBuffer[1] * 14 - 10;
```
What about reversing the assignment with something like:
```
uint8_t uuidLen = responseBuffer[1] == 1 ? 2 : 16;
uint16_t lengthPerDescriptor = uuidLen + 2;
```

carterd reviewed (commented):
Jeez missed this ... looks fine
